### ISSUE TITLE
Explosion when pasting to nested element

### ISSUE THREAD

**Issue by oskarwrobel** [ckeditor/ckeditor5#2010]
When pasting content to the document where the selection starts at the beginning of the nested element and ends in the next block

```html
<blockQuote>
    <paragraph>[Foo</paragraph>
</blockQuote>
<paragraph>Bar]</paragraph>
```

the selected content is removed (the entire blockquote) and then new content is inserted at the selection position. The problem is the selection position does not change after deleting content and still refers to the removed place in the document (path `[0,0,0]` or `[ '$root', 'blockQuote', 'paragraph' ]`). When comes to insertion there is an explosion because insert position does not exist.

We should manually fix selection between deleting and inserting content. Model selection post-fixer could be extracted to som util (or maybe it is already) and called manually.

![Aug-30-2019 11-14-20](https://user-images.githubusercontent.com/1057056/64010078-58720880-cb19-11e9-8e5f-04c209a14f57.gif)

```
ckeditorerror.js:66 Uncaught CKEditorError: model-position-path-incorrect: The position's path is incorrect. Read more: https://ckeditor.com/docs/ckeditor5/latest/framework/guides/support/error-codes.html#error-model-position-path-incorrect
 {"position":{"root":"main","path":[0,0,0],"stickiness":"toNext"}}
    at Position.get parent [as parent] (http://localhost:8125/ckeditor5-block-quote/tests/manual/blockquote.js:59068:11)
    at new Insertion (http://localhost:8125/ckeditor5-block-quote/tests/manual/blockquote.js:65021:48)
    at Object.callback (http://localhost:8125/ckeditor5-block-quote/tests/manual/blockquote.js:64941:21)
    at Model._runPendingChanges (http://localhost:8125/ckeditor5-block-quote/tests/manual/blockquote.js:53260:58)
    at Model.change (http://localhost:8125/ckeditor5-block-quote/tests/manual/blockquote.js:52682:16)
    at insertContent (http://localhost:8125/ckeditor5-block-quote/tests/manual/blockquote.js:64924:15)
    at Model.insertContent (http://localhost:8125/ckeditor5-block-quote/tests/manual/blockquote.js:52869:79)
    at Model.<anonymous> (http://localhost:8125/ckeditor5-block-quote/tests/manual/blockquote.js:113897:32)
    at Model.fire (http://localhost:8125/ckeditor5-block-quote/tests/manual/blockquote.js:111772:29)
    at Model.<computed> [as insertContent] (http://localhost:8125/ckeditor5-block-quote/tests/manual/blockquote.js:113901:16)
```

**Comment by oskarwrobel** [comment]
Turns out fix is very easy. Insertion position is stored before deleting content but should be after.

```diff
-               const insertionPosition = selection.getFirstPosition();
-
                if ( !selection.isCollapsed ) {
                        model.deleteContent( selection, { doNotAutoparagraph: true } );
                }

+               const insertionPosition = selection.getFirstPosition();
                const insertion = new Insertion( model, writer, insertionPosition );```

**Comment by oskarwrobel** [comment]
Unfortunately, fix I proposed may lead to losing target position.

For the following model
```html
<blockQuote>
    <p>Foo</p>
</blockQuote>
<p>Bar</p>
```
insert `abc` content in the place of `<p>Bar</p>`

The expected result will be:
```html
<blockQuote>
    <p>Foo</p>
</blockQuote>
<p>abc</p>
```
The actual result:
```html
<blockQuote>
    <p>Fooabc</p>
</blockQuote>
```
After deleting content the selection moves to the end of the last block element wher` $text` is allowed.

**Comment by Reinmar** [comment]
Fixed with ckeditor/ckeditor5-engine#1830.
